src/odemis/acq/stream/_sync.py

```diff
@@ -1783,7 +1783,12 @@ def _onCompletedData(self, n, raw_das):
         for d in raw_das:
             d.metadata[MD_DESCRIPTION] = sname
 
-        if len(raw_das) != numpy.prod(self.repetition.value):
+        # if polarization analyzer HW: can be 6 images per beam pos
+        if hasattr(self._streams[1], "analyzer") and self._streams[1].acquireAllPol.value == True:
+            num_images = 6
```

Comment:
Don't hardcode `6`, but use `len(POL_POSITIONS)`, that avoid giving it a feeling of a "magical constant". As a bonus, if for some reason the POL_POSITIONS changes, it will be updated automatically.